### PR TITLE
Add a package to PGP sign data.

### PR DESCRIPTION
**Pull request by dlorenc**
This is not used yet, but will be hooked up to the controller in a follow up PR.
This also includes test data in the same format we expect from the actual Kubernetes secret.

### CODE REVIEW

File: pkg/signing/pgp/testdata/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Real name: Tekton Unit Tests
+Email address: testing@tekton.dev
+You selected this USER-ID:
+    "Tekton Unit Tests <testing@tekton.dev>"
+$ gpg --export "Tekton Unit Tests" > pgp.private-key
+$ gpg --export-secret-key "Tekton Unit Tests" --armor > pgp.public-key
+$ echo -n "testing123" > pgp.passphrase
+```
+
+The passphrase is: `testing123`.
```
Comment (SantiagoTorres) [inline pkg/signing/pgp/testdata/README.md:19]: I wonder if we could use a batch gpg call to generate this with a script we can store in this directory.

File: pkg/signing/pgp/pgp.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		logger.Debugf("Found passphrase for private key at %s. Decrypting...", passphrasePath)
+		passphrase, err = ioutil.ReadFile(passphrasePath)
+		if err != nil {
+			return nil, err
+		}
+		if err := s.key.PrivateKey.Decrypt(passphrase); err != nil {
+			return nil, err
+		}
+	} else {
+		logger.Debugf("No passphrase foudn for private key at  %s", passphrasePath)
```
[the remark this answers is not in the document]
Comment (dlorenc) [reply pkg/signing/pgp/pgp.go]: Thanks!

**Comment by dlorenc** [comment]
> Thanks @dlorenc ! Should we introduce some sort of SignerType that is configurable based on the same ConfigMap that defines the signing format and storage backends that are enabled?

Yeah I think so! I've tried to start adding support for a configmap following the patterns in knative/pkg/configmap, but gotten lost a few times. I think the next step is to setup some kind of config, then wire it all up.

**Pull request merged**